### DISCUSSION TITLE
Crud Service Unknown Field Error

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#476] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12.0.2

### Description

Hi,
I have the crud service (v6.10.0) given a GET request with the _q field evaluated like this:
```
_q={"$and":[{"$or":[{"name":{"$in":["name1","name2"]}},{"managementLiv1":{"$in":[]}}]}]}
```
that is returning this error:
```
{
   "err":{
      "type":"Error",
      "message":"Unknown field: managementLiv1",
      "stack":"Error: Unknown field: managementLiv1\n    at traverse (/home/node/app/lib/QueryParser.js:253:13)\n    at QueryParser.traverseBinded (/home/node/app/lib/QueryParser.js:48:7)\n    at Array.forEach (<anonymous>)\n    at traverse (/home/node/app/lib/QueryParser.js:240:20)\n    at QueryParser.traverseBinded (/home/node/app/lib/QueryParser.js:48:7)\n    at Array.forEach (<anonymous>)\n    at traverse (/home/node/app/lib/QueryParser.js:240:20)\n    at QueryParser.traverseBinded (/home/node/app/lib/QueryParser.js:48:7)\n    at Array.forEach (<anonymous>)\n    at traverse (/home/node/app/lib/QueryParser.js:240:20)"
   },
   "msg":"Unknown field: managementLiv1"
}
```

Trying the same query directly on mongoDB is returning the results correctly. Also the field in the MongoDB Crud section of the console is defined.

What could possibly be the reason that I'm getting this error?

### Actual Outcome

Error Unknown field

### Expected Outcome

The result fields

**Comment by fredmaggiowski** [comment]
Hi, is the field `managementLiv1` present within the MongoDB Crud collection you have defined in Console?

**Reply by elmalakomar** [reply]
yes it is defined as an array of string

**Comment by danibix95** [comment]
Hi, could you please provide the definition of the `managementLiv1` field?  
Is it a root level property (defined as an array) or it is a nested field with a proper JSON Schema definition (`"type": "array"`)?

**Reply by elmalakomar** [reply]
Hi @danibix95 ,
it is a root level property defined as array of strings.

**Reply by danibix95** [reply, marked as answer]
Hi! The problem is due to an internal logic of CRUD Service that prevents the usage of fields names containing digits within the `_q` query parameter. This issue has been fixed in this [PR](https://github.com/mia-platform/crud-service/pull/297). Soon a new image with tag `main` will be available to be tested.

Beware that it might require you to update to the next major version of CRUD Service.
